rmf_schedule_visualizer/src/VisualizerData.cpp

```diff
@@ -222,4 +248,46 @@ std::vector<std::vector<
   return conflicts;
 }
 
+std::vector<Element> VisualizerDataNode::get_negotiation_trajectories(
+  uint64_t conflict_version, const std::vector<uint64_t>& sequence) const
+{
+  std::vector<Element> trajectory_elements;
+
+  auto table_view = _negotiation->table_view(conflict_version, sequence);
+  if (!table_view)
+  {
+    RCLCPP_WARN(
+      this->get_logger(), "table_view for conflict %d not found!",
+      conflict_version);
+    return trajectory_elements;
+  }
+
+  rmf_traffic::RouteId route_id = 0;
+  auto add_route = [&](rmf_traffic::ConstRoutePtr route_ptr,
```

Comment:
Similar to my earlier comment, it's best practice to explicitly capture all the variables. In this case that would look like `[&route_id, &table_view, &trajectory_elements]`.

However, this lambda is a bit different from the others, because this lambda's entire lifetime is spent inside of this function scope. That makes the risk of leaks or undefined behavior much much smaller. Still, I would suggest following best practices.

Reply:
adjusted captures 3b1c580